presto-main/src/main/java/io/prestosql/operator/aggregation/builder/SpillableHashAggregationBuilder.java

```diff
@@ -173,6 +188,21 @@ private boolean shouldMergeWithMemory(long memorySize)
     public WorkProcessor<Page> buildResult()
     {
         checkState(hasPreviousSpillCompletedSuccessfully(), "Previous spill hasn't yet finished");
+        producingOutput = true;
+
+        // convert revocable memory to user memory as memory must not be revoked when aggregation operator is finishing
+        if (localRevocableMemoryContext.getBytes() > 0) {
+            long currentRevocableBytes = localRevocableMemoryContext.getBytes();
+            localRevocableMemoryContext.setBytes(0);
+            if (!localUserMemoryContext.trySetBytes(hashAggregationBuilder.getSizeInMemory())) {
+                // TODO: this might fail (even though we have just released memory), but we don't
+                // have a proper way to atomically convert memory reservations
+                localRevocableMemoryContext.setBytes(currentRevocableBytes);
```

Comment:
for `trySetBytes` you were passing `hashAggregationBuilder.getSizeInMemory()`, here `currentRevocableBytes`.

i would use `currentRevocableBytes + localUserMemoryContext.getBytes()` in both places, so that the code in this method does not need to know how memory footpring is calculated. Let's leave this to `updateMemory()`.